worldwind/src/commonMain/kotlin/earth/worldwind/WorldWind.kt

```diff
@@ -142,6 +142,9 @@ open class WorldWind @JvmOverloads constructor(
 
         // Reset screen density factor.
         densityFactor = 1f
+
+        // Clear any cached OpenGL resources and state, which are now invalid.
+        dc.contextLost()
```

Comment:
I added this line of code because I saw that dc.contextLost() is only called after initialization and not during destruction, so this may cause a memory leak because the openGL buffers were not removed, instead we were creating a new dc.

Reply:
Interesting why original NASA engine has contextLost only in surfaceCreted phase https://github.com/NASAWorldWind/WorldWindAndroid/blob/0dacfec3da57b42a5228d99ca862a6ec201e2800/worldwind/src/main/java/gov/nasa/worldwind/WorldWindow.java#L729

Is it ok to cleare DC related variables not from an OpenGL sthread? As reset is called from main thread.

Reply:
I am not sure if we need to call this in reset fro Android iplementation, but probably we need to make separete method which will contain this dc.contextLost and call it in Web WorlWindow implementation in contextLost method? What do you think?

        /**
         * Called when the surface is created or recreated.
         * <p>
         * Called when the rendering thread
         * starts and whenever the EGL context is lost. The EGL context will typically
         * be lost when the Android device awakes after going to sleep.
         * <p>
         * Since this method is called at the beginning of rendering, as well as
         * every time the EGL context is lost, this method is a convenient place to put
         * code to create resources that need to be created when the rendering
         * starts, and that need to be recreated when the EGL context is lost.
         * Textures are an example of a resource that you might want to create
         * here.
         * <p>
         * Note that when the EGL context is lost, all OpenGL resources associated
         * with that context will be automatically deleted. You do not need to call
         * the corresponding "glDelete" methods such as glDeleteTextures to
         * manually delete these lost resources.
         * <p>
         * @param gl the GL interface. Use <code>instanceof</code> to
         * test if the interface supports GL11 or higher interfaces.
         * @param config the EGLConfig of the created surface. Can be used
         * to create matching pbuffers.
         */
        void onSurfaceCreated(GL10 gl, EGLConfig config);

Reply:
Do we need to free buffer manually on context lost at all? As I understand all opengl resources die autmatically on context lost.